kernel/packages/unity-interface/loader.ts

```diff
@@ -0,0 +1,105 @@
+import future from 'fp-future'
+import { parseUrn } from '@dcl/urn-resolver'
+
+declare const globalThis: any
+
+export type DclRenderer = {}
+export type UnityLoader = {}
+export type LoadRendererResult = {
+  DclRenderer: DclRenderer
+  UnityLoader: UnityLoader
+  buildConfigPath: string
+}
+
+/**
+ * This function is used by preview, instead of using "." as root,
+ * preview uses '/@/artifacts'
+ */
+function getRendererArtifactsRoot() {
+  if (typeof globalThis.RENDERER_ARTIFACTS_ROOT == 'undefined') {
+    throw new Error('RENDERER_ARTIFACTS_ROOT is undefined')
+  } else {
+    return new URL(globalThis.RENDERER_ARTIFACTS_ROOT, document.location.toString()).toString()
+  }
+}
+
+// TODO: return type DclRenderer
+async function injectRenderer(baseUrl: string): Promise<LoadRendererResult> {
+  const scriptUrl = new URL('index.js', baseUrl).toString()
+  const buildConfigPath = new URL('unity.json', baseUrl).toString()
```

Comment:
Since Unity 2020 we have to provide the loader configuration manually. If loading `unity.json` is the only way to provide the loader config we may have to port this code over.

Reply:
Can we move that a build with that version into a unity-renderer repo branch? It will be easier to test if we already have it there, and we could do this PR one time only

Reply:
But this code will keep living on this repo, right? This is to cover the case of having kernel with a custom renderer build. So we'll have to do this PR only once in any case. (and patch it anyways after the 2020 renderer is live).

I think we can just merge this as-is and patch it afterwards. Bear in mind that we are somehow going to lose compatibility with pre-2020 PRs after we do that. However, the version change will be around next week, so this should be a non-issue.

Reply:
OK, please point me to somewhere where I can learn about this change, will merge without it